File: leetcode/python/58_length_of_last_word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        r = len(s) - 1
        while r >= 0:
            if s[r] == " ":
                r -= 1
                continue
            
            l = r
            while l >= 0 and s[l] != " ":
                l -= 1
            return r - l
        return 0

File: leetcode/rust/src/58_length_of_last_word.rs
impl Solution {
    pub fn length_of_last_word(s: String) -> i32 {
        let mut count = 0;
        let mut last = 0;
        for c in s.chars() {
            if c == ' ' {
                if count > 0 {
                    last = count;
                }
                count = 0;
            } else {
                count += 1;
            }
        }
        if count > 0 {
            return count;
        }
        last
    }
}